code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -3678,5 +3678,102 @@
+	. = ..()
+	if(!(methods & (INGEST|INJECT|PATCH)) || !iscarbon(exposed_mob))
+		return
+
+	var/mob/living/carbon/exposed_carbon = exposed_mob
+	var/obj/item/organ/internal/stomach/ethereal/stomach = exposed_carbon.get_organ_slot(ORGAN_SLOT_STOMACH)
+	if(istype(stomach))
+		stomach.adjust_charge(reac_volume * 3)
+
+/datum/reagent/consumable/ethanol/telepole
+	name = "Telepole"
+	description = "A grounding rod in the form of a drink.  Recharges etherials, and gives temporary shock resistance."
+	boozepwr = 50
+	color = "#b300ff"
+	quality = DRINK_NICE
+	taste_description = "the howling storm"
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/glass_style/drinking_glass/telepole
+	required_drink_type = /datum/reagent/consumable/ethanol/telepole
+	name = "Telepole"
+	desc = "A liquid grounding rod. Recharges etherials and grants temporary shock resistance."
+	icon = 'icons/obj/drinks/mixed_drinks.dmi'
+	icon_state = "telepole"
+
+/datum/reagent/consumable/ethanol/telepole/on_mob_metabolize(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+
+/datum/reagent/consumable/ethanol/telepole/on_mob_end_metabolize(mob/living/affected_mob)
+	REMOVE_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	return ..()
+
+/datum/reagent/consumable/ethanol/telepole/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume) //can't be on life because of the way blood works.
+	. = ..()
+	if(!(methods & (INGEST|INJECT|PATCH)) || !iscarbon(exposed_mob))
+		return
+
+	var/mob/living/carbon/exposed_carbon = exposed_mob
+	var/obj/item/organ/internal/stomach/ethereal/stomach = exposed_carbon.get_organ_slot(ORGAN_SLOT_STOMACH)
+	if(istype(stomach))
+		stomach.adjust_charge(reac_volume * 2)
+
+/datum/reagent/consumable/ethanol/pod_tesla
+	name = "Pod Tesla"
+	description = "Ride the lightning!  Recharges etherials, suppresses phobias, and gives strong temporary shock resistance."
+	boozepwr = 80
+	color = "#00fbff"
+	quality = DRINK_FANTASTIC
+	taste_description = "victory, with a hint of insanity"
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/glass_style/drinking_glass/pod_tesla
+	required_drink_type = /datum/reagent/consumable/ethanol/pod_tesla
+	name = "Pod Tesla"
+	desc = "Ride the lightning! Recharges etherials, suppresses phobias, and grants strong temporary shock resistance."
+	icon = 'icons/obj/drinks/mixed_drinks.dmi'
+	icon_state = "pod_tesla"
+
+/datum/reagent/consumable/ethanol/pod_tesla/on_mob_metabolize(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	ADD_TRAIT(affected_mob, TRAIT_TESLA_SHOCKIMMUNE, type)
+	ADD_TRAIT(affected_mob, TRAIT_FEARLESS, type)
+
+
+/datum/reagent/consumable/ethanol/pod_tesla/on_mob_end_metabolize(mob/living/affected_mob)
+	REMOVE_TRAIT(affected_mob, TRAIT_SHOCKIMMUNE, type)
+	REMOVE_TRAIT(affected_mob, TRAIT_FEARLESS, type)
+	REMOVE_TRAIT(affected_mob, TRAIT_TESLA_SHOCKIMMUNE, type)
```

Comment:
use `remove_traits()` instead. you could also get away with having a list of these traits on the reagent as well to slim down on copypasta